pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -144,14 +145,16 @@ func (r *ReconcileChangeTierRequest) changeTier(log logr.Logger, changeTierReque
 	namespaces := make([]toolchainv1alpha1.NSTemplateSetNamespace, len(nsTemplateTier.Spec.Namespaces))
 	for i, ns := range nsTemplateTier.Spec.Namespaces {
 		namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
-			Type:     ns.Type,
-			Revision: ns.Revision,
+			Type:        ns.Type,
+			Revision:    ns.Revision,
+			TemplateRef: nstemplatetiers.NewTierTemplateName(changeTierRequest.Spec.TierName, ns.Type, ns.Revision), // link to the TierTemplate resource, whose name is: `<tierName>-<nsType>-<revision>`,
```

Comment:
The same thing again:
```suggestion
			TemplateRef: ns.TemplateRef,
```